server/app/routes/api/v1/va/ambassadors.js

```diff
@@ -243,40 +191,16 @@ async function signup(req, res) {
   req.body.externalId = req.externalId;
   let new_ambassador = null;
 
-  //check carrier lookup for blocked carriers
-  let carrierLookup = await carrier(normalize(req.body.phone));
-  if(carrierLookup.carrier.isBlocked) {
-    return error(400, res, `We're sorry, due to fraud concerns '${carrierLookup.carrier.name}' phone numbers are not permitted. Please try again.`, { request: req.body });
+  const carrierLookup = await validateCarrier(req.body.phone);
+  const { carrier: { name: carrierName, isBlocked } } = carrierLookup;
+  if (isBlocked) {
+    return error(400, res, `We're sorry, due to fraud concerns '${carrierName}' phone numbers are not permitted. Please try again.`, { request: req.body });
   }
 
-  // check against Twilio caller ID and Ekata data
-  let twilioCallerId = await caller_id(req.body.phone);
-  let ekataReversePhone = await reverse_phone(req.body.phone);
-  let verification = [];
-
-  if (twilioCallerId) {
-    try {
-      verification.push({
-        source: 'Twilio',
-        name: twilioCallerId
-      })
-    } catch (err) {
-      logger.error("Could not get verification info for ambassador: %s", err);
-    }
-  }
-  if (ekataReversePhone) {
-    try {
-      verification.push({
-        source: 'Ekata',
-        name: ekataReversePhone.addOns.results.ekata_reverse_phone
-      })
-    } catch (err) {
-      logger.error("Could not get verification info for ambassador: %s", err);
-    }
-  }
+  const verifications = verifyCallerIdAndReversePhone(req.body.phone);
```

Comment:
Ambassadors, unlike Triplers, will only have one verification. It's not possible (currently, anyway) to have more than one. We should call this `verification` for clarity, imho.

Reply:
and another note, down below a few more lines, we have ```return error(500, res, 'Unable to update ambassador form data', { ambassador: req.body, verification: verification });```